### PR TITLE
Document unstable linker flags in the unstable book

### PR DESCRIPTION
**Pull request by lqd**
This PR:
- removes the unstable linker flavors from the stable documentation
- documents the unstable `-C linker-flavor` values in the unstable book (including the ones previously described as stable)
- documents the unstable `-C link-self-contained` components in the unstable book

I don't know if these have some common reviewers, but if not r? @petrochenkov as the most knowledgeable person in this area. Please feel free to reassign, I know that you are quite busy.

### CODE REVIEW

File: src/doc/unstable-book/src/compiler-flags/codegen-options.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+(everything or nothing), but there exists a set of unstable values with finer-grained control,
+`-Clink-self-contained` can accept a comma-separated list of components, individually enabled
+(`+component`) or disabled (`-component`):
+- `crto`: CRT objects (e.g. on `windows-gnu`, `musl`, `wasi` targets)
+- `libc`: libc static library (e.g. on `musl`, `wasi` targets)
+- `unwind`: libgcc/libunwind (e.g. on `windows-gnu`, `fuchsia`, `fortanix`, `gnullvm` targets)
+- `linker`: linker, dlltool, and their necessary libraries (e.g. on `windows-gnu` and for
+  `rust-lld`)
+- `sanitizers`: sanitizer runtime libraries
+- `mingw`: other MinGW libs and Windows import libs
```
Comment (petrochenkov) [inline src/doc/unstable-book/src/compiler-flags/codegen-options.md:47]: Probably makes sense to mention that none of these, except `linker`, are actually implemented (besides option parsing).

**Comment by petrochenkov** [comment]
r=me with https://github.com/rust-lang/rust/pull/119024#discussion_r1429312337 addressed.
@rustbot author

File: src/doc/unstable-book/src/compiler-flags/codegen-options.md (hunk 1 above)
Comment (lqd) [reply src/doc/unstable-book/src/compiler-flags/codegen-options.md:47]: I brushed on this just below, but can say that they’re not yet implemented yes.

Comment (lqd) [reply src/doc/unstable-book/src/compiler-flags/codegen-options.md:47]: Done

**Pull request merged**